test/ci_workflow_gen.go

```diff
@@ -125,7 +127,7 @@ type unitTest struct {
 }
 
 type clusterTest struct {
-	Name, Shard                  string
+	Name, Shard, Platform        string
```

Comment:
This is elegant.
Only one nit, `Platform` doesn't seem like the right term for database (hat is what we are changing here).
How about using `Flavor` since we use that elsewhere in the code base?

Reply:
I had similar thoughts, but:
1. We already use Platform this way (e.g. mysql57 or mysql8) for the unit tests and self hosted [in the same file](https://github.com/vitessio/vitess/blob/9f6bd06b5142ef4faa9b909879a448368b4d1657/test/ci_workflow_gen.go) 
2. We already use OS, and Flavor is used for the GTID type so the Flavor is mysql56 for 5.7 and 8.0 in those contexts

I'm fine changing it to whatever though.

Reply:
alright, you've convinced me :)
If we are already using Platform, it makes sense to keep it consistent.